yarn-project/noir-compiler/src/compile/noir/file-manager/file-manager.ts

```diff
@@ -126,4 +134,23 @@ export class FileManager {
   #getPath(name: string) {
     return isAbsolute(name) ? name : join(this.#dataDir, name);
   }
+
+  /**
+   * Reads a file from the filesystem
+   * @param dir - File to read
+   * @param options - Readdir options
+   */
+  public async readdir(
+    dir: string,
+    options?: {
+      /**
+       * Traverse child directories recursively
+       */
+      recursive: boolean;
+    },
+  ) {
+    const dirPath = this.#getPath(dir);
+    const files = await this.#fs.readdir(dirPath, options);
+    return files.map(file => path.join(dirPath, file));
+  }
```

Comment:
The only thing I'd point out is the edge case that the contract folder also contains a `node_modules` or something and we end up traversing a super deep hierarchy.

I think it's fine for now though, but something to be aware of.